.circleci/config.yml

```diff
@@ -374,23 +374,20 @@ jobs:
     executor: ubuntu-machine-executor
     steps:
       - checkout
-      - *attach_generated_sql
-      - *copy_generated_sql
       - when:
           condition: &deploy-condition
             or:
               - equal: [main, << pipeline.git.branch >>]
               - << pipeline.git.tag >>
           steps:
+            - *attach_generated_sql
+            - *copy_generated_sql
             - docker/check:
                 docker-password: DOCKER_PASS
                 docker-username: DOCKER_USER
-      - docker/build: &public-image
-          image: ${CIRCLE_PROJECT_USERNAME+$CIRCLE_PROJECT_USERNAME/}${CIRCLE_PROJECT_REPONAME:-bigquery-etl}
-          tag: ${CIRCLE_TAG:-latest}
```

Comment:
It seems unnecessary to build the docker image for every PR. It also takes around 5 minutes. It should be fine just doing it on main or am I missing something here?

Reply:
I assume the intent here was to catch if we've introduced any errors that will affect the stability of building the image. I agree it's a pretty expensive step that would be nice to skip, though.

Reply:
klukas has it right. I'm fine with this change.